components/channel_header/channel_header.tsx

```diff
@@ -589,9 +597,9 @@ class ChannelHeader extends React.PureComponent {
                                     id='channel_header.editLink'
                                     defaultMessage='Edit'
                                 >
-                                    {(message) => (
+                                    {(message: string[]) => (
```

Comment:
This is actually incorrect. The type of message is `React.ReactNode[]` because of the rich text formatting stuff that `FormattedMessage` can do.

Instead of that, I think you can replace this whole `FormattesMessage` with a `LocalizedIcon` component. That'll also be a bit simpler to read code-wise

Reply:
I didn't see any change in the behavior when I use `string[]` but using `LocalizedIcon` is better option and chance to simplify the code. Therefore, I updated it to use `LocalizedIcon`

Reply:
Thanks for changing that! I think it's fine to make that other change here as well.

I'm surprised it type checked fine before this since while those functions usually receive strings, they can return other React nodes if we were to use [rich text formatting](https://formatjs.io/docs/react-intl/components/#rich-text-formatting) as part of them. I guess they've done some stuff to make it so that the child function types correctly in simpler cases like this

Reply:
Thanks updated other file as well.